ArjunaCore/arjuna/tests/classes/com/hp/mwtests/ts/arjuna/performance/Performance1.java

```diff
@@ -22,8 +22,9 @@ public void test() {
         int threadCount =  1;
         int batchSize = 100;
 
+        // Using 10 minutes timeout (maxTestTime), it should be lower than 1 minute
         Measurement measurement = new Measurement.Builder(getClass().getName() + "_test1")
-                .maxTestTime(0L).numberOfCalls(numberOfTransactions)
+                .maxTestTime(600000L).numberOfCalls(numberOfTransactions)
```

Comment:
The timeout controls how long to wait for all threads to reach the barrier. But all threads should reach the barrier and if one of them doesn't reach it then that is the bug ie you need to find out which thread is blocked from hitting the barrier - setting a non zero timeout is bypassing the problem.

Also note that the measurement framework was written before JMH was available and the tests that use it should be rewritten to use JMH, there aren't many of them left so that should be our priority after which we can delete the framework.

Reply:
IIUC Marco spotted this (at least most recently) when databases weren't working so basically the test was not going to be successful (presumably for any thread basically). I understand what you are saying though and I think it is possible to consider revering these timeouts (even if I am not super comfortable with that as they give a bit more protection that what I am about to say).

So following on from that, @marcosgopen please can you confirm if the addition (in the Measurement.java elsewhere in your changeset) of one of the new `throw new RuntimeException(e);` would handle the same situation the same anyway and thus the timeouts don't make a material improvement and could therefore be reverted? Was it the BrokenBarrierException clause?

Reply:
@tomjenkinson is right @mmusgrov , the problem is that the database returned an error and the test hung instead of failing.

Reply:
> @marcosgopen please can you confirm if the addition (in the Measurement.java elsewhere in your changeset) of one of the new throw new RuntimeException(e); would handle the same problem as the timeouts anyway and thus the timeouts don't make a material improvement and could therefore be reverted? In other words (and from a bit of presumption), was it the BrokenBarrierException clause that would be considered the more direct fix when the database is not working?

I see what you mean @tomjenkinson, I think you are right. Probably the fact that the exception is caught and not re thrown is the problem. So I can remove the changes to the Performance* classes and keep the Measurement change only. That would be enough to prevent the test from hanging. Thanks

Reply:
@tomjenkinson and @marcosgopen  Why don't you just port them to JMH and move them to the perf repo and then delete the measurement framework? Porting these perf tests isn't particularly difficult.

Reply:
We should do that @mmusgrov I agree. It seems a good next step in this particular area (I think fixing possible hangs is still a good win in the meantime).

For the porting side of things, I think I found the issue [https://issues.redhat.com/browse/JBTM-2621] - I think it needs re-opening as I think it might not have been completed from a quick look at the linked pull request, do you agree with my read or do you think we need a new issue? The hanging test was `com.hp.mwtests.ts.arjuna.performance.Performance1` which is different to the ones mentioned explicitly in JBTM-2621 but then JBTM-2621 talks about `if there are no others then once ported the old performance benchmark harness can be removed. `